common/src/main/java/org/apache/comet/parquet/NativeBatchReader.java

```diff
@@ -421,9 +515,40 @@ public void init() throws Throwable {
       CometFileKeyUnwrapper keyUnwrapper = null;
       if (encryptionEnabled) {
         keyUnwrapper = new CometFileKeyUnwrapper();
-        keyUnwrapper.storeDecryptionKeyRetriever(file.filePath().toString(), conf);
+        keyUnwrapper.storeDecryptionKeyRetriever(filePath, conf);
       }
 
+      // Filter out columns with preinitialized readers from sparkSchema before making the
+      // call to native
+      if (preInitializedReaders != null) {
+        StructType filteredSchema = new StructType();
+        StructField[] sparkFields = sparkSchema.fields();
+        // List<Type> fileFields = fileSchema.getFields();
+        for (int i = 0; i < sparkFields.length; i++) {
+          // Keep the column if:
+          // 1. It doesn't have a preinitialized reader, OR
+          // 2. It has a preinitialized reader but exists in fileSchema
+          boolean hasPreInitializedReader =
+              i < preInitializedReaders.length && preInitializedReaders[i] != null;
+          int finalI = i;
+          boolean existsInFileSchema =
+              fileFields.stream().anyMatch(f -> f.getName().equals(sparkFields[finalI].name()));
+
+          if (!hasPreInitializedReader || existsInFileSchema) {
+            filteredSchema = filteredSchema.add(sparkFields[i]);
+          }
+        }
+        sparkSchema = filteredSchema;
+      }
+
+      //      String timeZoneId = conf.get("spark.sql.session.timeZone");
+      String timeZoneId = "UTC";
```

Comment:
Is this intentional ?
If it is then either move the comment below one line up or add a new comment why `timeZoneId` should be also always UTC. The commented out `conf.get("spark.sql.session.timeZone");` could be removed too.